outline/apple/tun2socks.go

```diff
@@ -64,3 +67,20 @@ func ConnectShadowsocksTunnel(tunWriter TunWriter, client *shadowsocks.Client, i
 	}
 	return outline.NewTunnel(client, isUDPEnabled, tunWriter)
 }
+
+// ConnectTunnel reads packets from a TUN device and routes it to a proxy server.
+// Returns an OutlineTunnel instance that should be used to input packets to the tunnel.
+//
+// `tunWriter` is used to output packets to the TUN (VPN).
+// `client` is the proxy client (created by [proxy.NewClient]).
+// `isUDPEnabled` indicates whether the tunnel and/or network enable UDP proxying.
+//
+// Sets an error if the tunnel fails to connect.
+func ConnectProxyTunnel(tunWriter TunWriter, client *proxy.Client, isUDPEnabled bool) (OutlineTunnel, error) {
```

Comment:
Not for this PR, but I wish the `isUDPEnabled` wasn't here. I believe it can be in the client logic instead (or in a tun2socks UDP handler, perhaps we should be handling UDP handlers instead.)